### PR TITLE
Wb 376

### PR DESCRIPTION
**Pull request by pierlucatonioloreply**
Create Query Users API

Follow the idea of the Get Policies with query, in the broker domain, file get-policies.go

### CODE REVIEW

File: user/get-users.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+type GetUsersReq struct {
+	Queries []struct {
+		Field string      `json:"field"`
+		Op    string      `json:"op"`
+		Value interface{} `json:"value"`
+		Type  string      `json:"type"`
+	} `json:"queries,omitempty"`
+	Limit int `json:"limit"`
+	Page  int `json:"page"`
+}
```
Comment (nahidalgo) [inline user/get-users.go]: This is equal to the `GetPoliciesReq`. I think there should be a Query struct in models. What do you think @Yousef-Hammar  @diogo-wopta

Comment (nahidalgo) [inline user/get-users.go]: Add these structs on the top of the file

File: user/user.go [hunk 2]
```diff
@@ -81,6 +81,12 @@ func User(w http.ResponseWriter, r *http.Request) {
 				Method:  http.MethodPatch,
 				Roles:   []string{models.UserRoleAdmin},
 			},
+			{
+				Route:   "/user/v1",
```
Comment (nahidalgo) [inline user/user.go]: The domain already has user, so you can leave only `v1`
```suggestion
				Route:   "/v1",
```

**Review by nahidalgo — CHANGES_REQUESTED** [review]

File: user/user.go [hunk 3]
```diff
@@ -81,6 +81,12 @@ func User(w http.ResponseWriter, r *http.Request) {
...
+				Handler: GetUsersFx,
+				Method:  http.MethodPost,
+				Roles:   []string{models.UserRoleAll},
```
Comment (nahidalgo) [inline user/user.go]: Here, the role should be `models.UserRoleAdmin`

**Review by nahidalgo — DISMISSED** [review]

**Review by Yousef-Hammar — APPROVED** [review]

**Review by diogo-wopta — APPROVED** [review]

**Review by diogo-wopta — APPROVED** [review]

**Pull request merged**